primary_beam_correction.py

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/env python
+import argparse
+import logging
+import numpy as np
+from astropy import units as u
+from astropy.io import fits
+from astropy import wcs
+from astropy import coordinates
+
+
+def read_fits(pathfits):
+    '''
+    This functions reads in the fits file
+    Input: Path to the fits file
+    Output: Fits file object
+    '''
+    fl = fits.open(pathfits)
+    images = fl[0]
+    return images
+
+
+def get_positions(path):
+    '''
+    This function reads the fits file and returns the sky coordinate
+    of the phase centre.
+    Input: path to the fits file
+    Output: Sky coordinate of the phase centre
+    '''
+    images = read_fits(path)
+    # Get pointing centre of observation
+    phase_centre_ra = images.header['OBSRA']
+    phase_centre_dec = images.header['OBSDEC']
+    # Convert to astropy.coordinates.SkyCoord object
+    phase_center = coordinates.SkyCoord(phase_centre_ra, phase_centre_dec, unit=(u.deg, u.deg))
+    # Get positions of other sources
+    source_pos = wcs.WCS(images.header)
+    return phase_center, source_pos
+
+
+def radial_offset(phase_center, source_pos, row, col):
+    '''
+    This function gets the position in ra and dec of a pixel and determine its
+    radial offset from phase centre.
+    Input: phase center and source positions
+    output: A separation vector between phase centre and source positions in radians.
+    '''
+    p2w = source_pos.pixel_to_world(row, col, 0, 0)[0]
+    separation_rad = p2w.separation(phase_center).rad
+    return separation_rad
+
+
+def central_freq(path):
+    '''
+    Function to get central frequency of each frequency plane.
+    Input : image header
+    Output : List of central frequency in MHz of each frequency plane.
+    '''
+    images = read_fits(path)
+    c_freq_plane = []
+    for i in range(1, images.header['NSPEC']+1):
+        c_freq_plane.append(images.header['FREQ{0:04}'.format(i)])
+    return np.array(c_freq_plane)
+
+
+def beam_pattern(separation_rad, path, c_freq=None):
+    '''
+    Making beam pattern using the DEEP2 paper
```

Comment:
Say "Cosine-squared power approximation from Mauch et al. (2020)" instead of "DEEP2 paper"

Reply:
Done